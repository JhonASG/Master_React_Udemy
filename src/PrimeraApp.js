import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({new_text, saludo, subtitulo})=>{
    const texto = 'My first application';
    const num = 1+2;
    const booleana = false;
    const arreglo = [1,2,3,4,5]
    const objeto = {
        nombre: "Jhon Sossa",
        edad: 23
    };

    return (
        <>
            <h1>{new_text}</h1>
            <h2>{saludo}</h2>
            <h3>{subtitulo}</h3>
            <p>Texto: {texto}</p>
            <p>Numero: {num}</p>
            <p>Booleano: {booleana}</p>
            <p>Arreglo: {arreglo}</p>
            <pre><p>Objeto:</p> {JSON.stringify(objeto, null, 3)}</pre>
            <pre><p>Desglose objeto:</p> {JSON.stringify(`${objeto.nombre} tiene ${objeto.edad} a√±os`, null, 3)}</pre>
        </>
    );
};

PrimeraApp.propTypes = {
    new_text: PropTypes.number.isRequired,
    saludo: PropTypes.string.isRequired
};

PrimeraApp.defaultProps = {
    subtitulo: "Goku es el sayayin mas fuerte"
};

export default PrimeraApp;