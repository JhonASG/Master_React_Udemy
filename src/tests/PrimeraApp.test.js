import React from 'react';
import { render, screen } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";

describe('Pruebas en el componente <PrimeraApp />', () => {
    test('Debe mostrar el mensaje "Hola soy Veguetta".', () => {
        const saludo = "Hola soy Veguetta";
        const num = 123;

        // Se renderiza el componente para verificarlo
        render( <PrimeraApp new_text={num} saludo= {saludo} /> );

        // Se evalua si el texto que contiene la variable saludo se encuentra en el documento
        // Esto es una forma de probar los componentes pero posteriormente se va a utilizar Enzyme
        // expect(screen.getByText(saludo)).toBeInTheDocument();
        expect(screen.getByText(saludo)).toMatchSnapshot();
    });

    test('Debe mostrar un subtitulo enviado por props', () => {
        const saludo = "Hola soy Veguetta";
        const num = 123;
        const subTitle = "Goku es el sayayin mas fuerte";

        // Se renderiza el componente para verificarlo 
        render(<PrimeraApp new_text={num} saludo= {saludo}/>);

        // De esta manera estoy validando si en el documento el contenido en h3 es el mismo que el de la variable subTitle
        expect(screen.getByText(subTitle)).toContainHTML('h3');
    });

    // Configuración de Enzyme -> por el momento solo se encuentra para version <17 de React
    /*
    /// setupTests.js:

    import Enzyme from 'enzyme';
    import Adapter from 'enzyme-adapter-react-16'
    import {createSerializer} from 'enzyme-to-json'

    Enzyme.configure({ adapter: new Adapter() });
    expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

    /// Para instalar enzyme -> https://github.com/wojtekmaj/enzyme-adapter-react-17
        import { shallow } from 'enzyme';
        import '@testing-library/jest-dom';

        describe('Prueba', () => {
            test('Debe mostrar el mensajev "Hola soy Veguetta".', () => {
                /// shallow() -> Funciona como el render() de testing-library solo que permite simular
                /// clicks, etc.
                const wrapper = shallow(<PrimeraAPP new_text={num} saludo= {saludo} />);

                /// Se evalua que lo que muestra el componente este en la aplicación
                /// Instalar enzyme-to-json -> Con estos tenemos una fotografía de lo que se esta
                /// renderizando [npm install --save-dev enzyme-to-json]
                /// Snapshot -> No tocar la carpeta que se crea
                /// En el Snapshot guardo todos los cambios del componente renderizado.
                expect( wrapper ).toMatchSnapshot();
            });
        })
    */
});