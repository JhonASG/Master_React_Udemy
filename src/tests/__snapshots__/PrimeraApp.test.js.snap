// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <PrimeraApp /> Debe mostrar el mensaje "Hola soy Veguetta". 1`] = `
<h2>
  Hola soy Veguetta
</h2>
`;
